src/main/java/uk/gov/justice/probation/courtcaseservice/restclient/communityapi/mapper/OffenderMapper.java

```diff
@@ -59,6 +64,51 @@ private Conviction buildConviction(CommunityApiConvictionResponse conviction) {
                 .collect(Collectors.toList());
     }
 
+    public GroupedDocuments documentsFrom(CommunityApiGroupedDocumentsResponse documentsResponse) {
+        return GroupedDocuments.builder()
+            .documents(Optional.ofNullable(documentsResponse.getDocuments()).map(this::buildDocuments).orElse(Collections.emptyList()))
+            .convictions(Optional.ofNullable(documentsResponse.getConvictions()).map(this::buildConvictionDocuments).orElse(Collections.emptyList()))
+            .build();
+    }
+
+    private List<ConvictionDocuments> buildConvictionDocuments(List<CommunityApiConvictionDocuments> convictionDocuments) {
+        return convictionDocuments.stream()
+                .map(this::buildConvictionDocument)
+                .collect(Collectors.toList());
+    }
+
+    private ConvictionDocuments buildConvictionDocument(CommunityApiConvictionDocuments convictionDocuments) {
+        return ConvictionDocuments.builder()
+            .convictionId(convictionDocuments.getConvictionId())
+            .documents(Optional.ofNullable(convictionDocuments.getDocuments()).map(this::buildConvictionIdDocuments).orElse(Collections.emptyList()))
+            .build();
+    }
+
+    private List<OffenderDocumentDetail> buildConvictionIdDocuments(List<CommunityApiOffenderDocumentDetail> documentDetails) {
+        return documentDetails.stream()
+            .map(this::buildOffenderDocumentDetail)
+            .collect(Collectors.toList());
+    }
+
+    private List<OffenderDocumentDetail> buildDocuments(List<CommunityApiOffenderDocumentDetail> details) {
+        return details.stream()
+            .map(this::buildOffenderDocumentDetail)
+            .collect(Collectors.toList());
+    }
+
+    private OffenderDocumentDetail buildOffenderDocumentDetail(CommunityApiOffenderDocumentDetail detail) {
+        return OffenderDocumentDetail.builder()
+            .alfrescoId(detail.getId())
```

Comment:
Are we certain that this is the alfrescoId, if so is there a specific reason  we're calling that  fact out here rather than  just referring to it as a `documentId`?

Reply:
It's called alfrescoId in the community API and I thought it would be clearer if we kept it like that. Calling it "id" as we do for everything else (for numbers and such) when this is clearly for alfresco and looks like a GUID seemed better.

Reply:
Actually the Community API field is called "id" on the JSON objects but it comes in like this 
`    private OffenderDocumentDetail.OffenderDocumentDetailBuilder offenderDocumentDetailBuilderOf(Document document) {
        return OffenderDocumentDetail
                .builder()
                .author(authorOf(document))
                .createdAt(createAtOf(document))
                .documentName(document.getDocumentName())
                .id(document.getAlfrescoId())
                .lastModifiedAt(document.getLastSaved());
    }`

Reply:
Renamed to documentId